### PR TITLE
Backend implementation of pinning message

### PR DESCRIPTION
**Pull request by JonasMuehlmann**
# Before this PR can be merged, the following checkboxes have to be ticked

- [x] This code belongs to the backend and is tested, or tests are not useful
- [x] New functions with more than 5 lines have docstrings
- [x] New code has been integrated with the `MessengerService` if appropiate
- [x] New code has been integrated with the `SignalRService` if appropiate
- [x] New functions or classes have been documented in the _Entwurfsdokumentation_
- [x] Changes introduce no commented out code or `Console.Writeline()` statements used for debugging

# Related issues:

### CODE REVIEW

**Comment by JayKim94** [comment]
Could you please update our main DB with the new models? Do you still have all the scripts to generate them?

**Comment by JonasMuehlmann** [comment]
> Could you please update our main DB with the new models? Do you still have all the scripts to generate them? @JonasMuehlmann

@JayKim94  It should be up-to-date.

File: Messenger/Messenger.Core/Services/ChannelService.cs [hunk 1]
```diff
@@ -129,5 +130,81 @@ public static async Task<Channel> GetChannel(uint channelId)
...
+
+            return await SqlHelpers.NonQueryAsync(query);
+        }
+
+        /// <summary>
+        /// Retrieve the pinned messages of a channel
+        /// </summary>
+        /// <param name="channelId">The channel to retrieve pinned messages from</param>
+        /// <returns>An enumerable of message objects representing the pinned messages</returns>
+        public static async Task<IEnumerable<Message>> RetrievePinnedMessages(uint channelId)
```
[the remark this answers is not in the document]
Comment (JayKim94) [reply Messenger/Messenger.Core/Services/ChannelService.cs:189]: I thought you would change it to `RetrievePinnedMessage` and return a single object?

**Review by JayKim94 — COMMENTED** [review]
[RetrievePinnedMessages](https://github.com/JonasMuehlmann/BIB_VPR/pull/230#discussion_r667924734)

File: Messenger/Messenger.Core/Services/ChannelService.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (JonasMuehlmann) [reply Messenger/Messenger.Core/Services/ChannelService.cs:189]: Wouldn't you agree that it is useful to have multiple pinned messages per channel?

**Review by JayKim94 — APPROVED** [review]

**Pull request merged**